homeassistant/components/generic_thermostat/climate.py

```diff
@@ -110,7 +127,10 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     hot_tolerance = config.get(CONF_HOT_TOLERANCE)
     keep_alive = config.get(CONF_KEEP_ALIVE)
     initial_hvac_mode = config.get(CONF_INITIAL_HVAC_MODE)
-    away_temp = config.get(CONF_AWAY_TEMP)
+    presets = {}
+    for (key, value) in CONF_PRESETS.items():
+        if value in config:
+            presets[key] = config.get(value)
```

Comment:
```suggestion
    presets = {k: config[value] for key, value in CONF_PRESETS.items() if value in config}
```

Reply:
Use `config[value]` instead of `config.get` since you are already validating that the key exists in `config`